app/models/thread.php

```diff
@@ -0,0 +1,66 @@
+<?php
+class Thread extends AppModel
+{
+    public $validation = array(
+        'title' => array(
+            'length' => array(
+                'validate_between', 1, 30,
+            ),
+        ),
+    );
+
+    public static function getAll()
+    {
+        $threads = array();
+        $db = DB::conn();
+        $rows = $db->rows('SELECT * FROM thread');
+        foreach ($rows as $row){
+            $threads[] = new Thread($row);
+        }
+        return $threads;
+    }
+
+    public static function get($id)
+    {
+        $db = DB::conn();
+        $row = $db->row('SELECT * FROM thread WHERE id = ?', array($id));
+        return new self($row);
+    }
+
+    public function getComments()
+    {
+        $comments = array();
+        $db = DB::conn();
+        $rows = $db->rows('SELECT * FROM comment WHERE thread_id = ? ORDER BY created ASC', array($this->id));
```

Comment:
this should be done in the comment model. You can revise it like this
```php
public function getComments()
{
    return Comment::getByThreadId($this->id);
}
```
in `models/comment.php`
```php
public static function getByThreadId($thread_id)
{
    $comments = array();
    $db = DB::conn();
    $rows = $db->rows('SELECT * FROM comment WHERE thread_id = ?', array($this->id));
    foreach ($rows as $row) {
        $comments[] = new self($row);
    }
    return $comments;
}
```